httpcore/_backends/auto.py

```diff
@@ -1,10 +1,13 @@
 from ssl import SSLContext
-from typing import Dict, Optional
+from typing import Optional
 
 import sniffio
 
+from .._types import TimeoutDict
 from .base import AsyncBackend, AsyncLock, AsyncSemaphore, AsyncSocketStream
-from .sync import SyncBackend, SyncLock, SyncSemaphore, SyncSocketStream
+
+# The following line is imported from the _sync modules
```

Comment:
Curious: why don't the `_sync/*.py` modules import from eg `_backends.sync` instead? Is it because of `unasync`?

Treating auto as "works both for sync and async" location seems a bit hacky. Should we instead make `_async` and `_sync` import from `_backends` directly, i.e. define an `__all__` in `_backends/__init__.py`?

(Not a comment on this PR in particular, just something we could address separately.)

Reply:
I don't really know why tbh, it did seem a bit odd but out of scope so I left it. Your suggestion sounds sensible though.